networkit/vizbridges.py

```diff
@@ -0,0 +1,335 @@
+		a Maxent-Stress layout is used. Option :code:`Dimension.TwoForcePlotly` forces a plot 
+		in 2D with using Plotly instead of Cytoscape. Default: networkit.vizbridges.Dimension.Two
+	nodeScores : list(float), optional
+		List of scores for each nodes, for example scores from a centrality measure. This is 
+		used for color-calculation of the nodes (continuous distribution). Provide either 
+		node_scores or node_partition - not both. Default: None
+	nodePartition : networkit.structures.Partition, optional
+		Partition object. This is used for color-calculation of the nodes (discrete distribution). 
+		Provide either node_scores or node_partition - not both. Default: None	
+	nodePalette : list(tuple(float, float, float)), optional
+		List consisting of normalized rgb-values. If none is given, seaborn.color_palette.colors is used. Default: None
+	showIds : boolean, optional
+		Set whether node ids should be visible in plot-widget. Default: True
+	customSize : int, optional
+		If not set, plugins will use a default size. Otherwise the widget will have a certain width and height. Default: None       
+	"""
+	# Sanity checks
+	if dimension == Dimension.Two and not hasCyto:
+		raise MissingDependencyError("ipycytoscape")
+
+	if (dimension == Dimension.Three or dimension == Dimension.TwoForcePlotly) and not hasPlotly:
+		raise MissingDependencyError("Plotly")
+
+	if nodeScores is not None:
+		if nodePartition is not None:
+			raise Exception("Provide either nodeScores or nodePartition - not both.")
+		if len(nodeScores) != G.upperNodeIdBound():
+			raise Exception("nodeScores should include scores for every node.")	
+
+	# Color palette is needed for node coloring with Plotly and Cytoscape
+	palette = _getColorPalette(nodePalette=nodePalette, nodePartition=nodePartition)
+	
+	# Set styling
+	if showIds:
+		s = [{
+			'selector': 'node',
+			'css': {
+				'background-color': 'data(color)',
+				'content': 'data(id)'}
+			}]
+	else:
+		s = [{
+			'selector': 'node',
+			'css': {
+				'background-color': 'data(color)'}
+			}]
+
+	if dimension == Dimension.Two:
+		# Create widget
+		graphWidget = ipycytoscape.CytoscapeWidget()
+
+		# Add data
+		nodes = []
+		edges = []
+		if G.isDirected():
+			edgeClass = "directed "
+		else:
+			edgeClass = "undirected "
+
+		# Color list (norm. RGB-values) is needed for node coloring with Cytoscape 
+		hcColors = _calculateNodeColoring(G, palette, nodeScores, nodePartition)
+
+		for i in G.iterNodes():
+			n = ipycytoscape.Node(data={"id": i, "color": hcColors[i] })						
+			nodes.append(n)
+
+		for u,v in G.iterEdges():
+			e = ipycytoscape.Edge(data={"source": u, "target": v, "classes": edgeClass })
+			edges.append(e)
+
+		# It is much faster to add edges and nodes in bulk.
+		graphWidget.graph.add_nodes(nodes)
+		graphWidget.graph.add_edges(edges, G.isDirected())
+
+		# Set layout
+		graphWidget.set_style(s)
+		graphWidget.set_layout(name='cose')
```

Comment:
Wouldn't be better to allow the user to choose the layout?

Reply:
Yeah I thought about that ... however the number of parameters are already long. Since the returned widget supports the same functions, the user can always change the layout (and other functions for both plugins) afterwards.

Reply:
Ah I see